back/engines/free/frontend/app/services/frontend/url_service.rb

```diff
@@ -79,21 +79,27 @@ def home_url(options = {})
       locale ? "#{url}/#{locale.locale_sym}" : url
     end
 
-    def verification_url(options = {})
-      pathname = options[:pathname]
+    def sso_return_url(options = {})
+      pathname = options[:pathname] || '/'
+      pathname = strip_existing_locale_from_path(pathname) if options[:locale]
       "#{home_url(options)}#{pathname}"
     end
 
     def signin_success_url(options = {})
```

Comment:
Is it still necessary to keep all these different methods when they all return the same URL?

Reply:
Good point, I was following the same pattern `verification_url`, but that does not need different methods either. Will change